action/flow/event/flowevents.go

```diff
@@ -0,0 +1,64 @@
+package event
+
+import (
+	"time"
+)
+
+type Status string
+
+const (
+	CREATED   = "Created"
+	COMPLETED = "Completed"
+	CANCELLED = "Cancelled"
+	FAILED    = "Failed"
+	SCHEDULED = "Scheduled"
+	SKIPPED   = "Skipped"
+	STARTED   = "Started"
+	WAITING   = "Waiting"
+	UNKNOWN   = "Created"
+)
+
+const FLOW_EVENT_TYPE = "flowevent"
+const TASK_EVENT_TYPE = "taskevent"
+
+// FlowEvent provides access to flow instance execution details
+type FlowEvent interface {
```

Comment:
FlowInput() is probably required if a user needs to log values on flow start

Reply:
@fm-tibco : Added FlowInput() in latest PR